sdk/src/assertion.rs

```diff
@@ -616,6 +612,10 @@ pub enum AssertionDecodeErrorCause {
 
     #[error(transparent)]
     CborError(#[from] serde_cbor::Error),
+
+    /// There was a problem decoding field.
+    #[error("the assertion had a mandatory field: {expected} that could not be decoded")]
+    FieldDecoding { expected: String },
```

Comment:
Can we rename this to something like `MissingField(String)`?

Reply:
The field could be there but in the wrong format, such as being an array or number rather than an object. MissingField would be incorrect in that case.

Reply:
Perhaps we could get more clear about the type that was expected.